### PR TITLE
meson: Some minor fixes/enhancements

### PR DESCRIPTION
**Pull request by mobin-2008**
Hi. After #263 (I'm waiting for @davmac314 review) It's time to post some minor fixes/patches for meson.
first commit is bugfix and other are enhancements. Each commit has its title (and its description about utmp related commit) for more info.

Regards

`Signed-off-by: Mobin "Hojjat" Aydinfar <mobin@mobintestserver.ir>`

### CODE REVIEW

**Comment by mobin-2008** [comment]
I removed my request for review because maybe you want to close #263 first. I opened this PR for CI checks mainly (CI checks in forks is painful!).

**Comment by davmac314** [comment]
> I opened this PR for CI checks mainly (CI checks in forks is painful!).

There's generally no need to run full CI until a PR is ready. You can test it locally until then. I would prefer you don't open PR until you believe it's ready (if it fails CI, it's fine to fix it then). No need to request review, I already get a notification that the PR is opened, the request for review is just another notification.

Pushing changes to PR once it's opened is best avoided because I may have already started reviewing. Then I have to start again or look separately at the changes between the different revisions. So please try to keep that to a minimum - get everything ready, check it over carefully, and then do one push when it's ready.

File: src/meson.build [hunk 1]
```diff
@@ -34,43 +34,37 @@ endif
...
-    'dinitctl.cc',
-    'options-processing.cc',
-    include_directories : default_incdir,
-    install : true,
-    install_dir : sbindir
-)
-executable(
-    'dinitcheck',
-    'dinitcheck.cc',
+foreach exec:['dinitctl', 'dinitcheck']
```
Comment (davmac314) [inline src/meson.build:41]: I think it's better to leave them separate, as it was. Later changes may add source files to one or the other and if you combine them now you just have to split them then, i.e. refactoring should not combine things that are _coincidentally_ the same.

You can use a `kwargs` argument and map with `include_directories`, `install` and `install_dir` values if you really want to remove some duplication.

**Review by davmac314 — CHANGES_REQUESTED** [review]
Looks ok except for one thing (noted).

In future though, please keep formatting changes and functional changes in separate PRs.

**Comment by mobin-2008** [comment]
> > I opened this PR for CI checks mainly (CI checks in forks is painful!).
> 
> There's generally no need to run full CI until a PR is ready. You can test it locally until then. I would prefer you don't open PR until you believe it's ready (if it fails CI, it's fine to fix it then). No need to request review, I already get a notification that the PR is opened, the request for review is just another notification.
> 
> Pushing changes to PR once it's opened is best avoided because I may have already started reviewing. Then I have to start again or look separately at the changes between the different revisions. So please try to keep that to a minimum - get everything ready, check it over carefully, and then do one push when it's ready.

I just reworded commit message of bugfix to include what error got fixed. I didn't change code.

**Comment by davmac314** [comment]
> I just reworded commit message of bugfix to include what error got fixed. I didn't change code.

I know, it's fine. I think I misunderstood what you meant when you said you opened it for CI checks mainly.

Anyway, it's fine to open multiple PRs, I will get to them when I have time. This one was easy to review so I did so.

**Review by davmac314 — APPROVED** [review]
Looks good, feel free to merge

**Comment by mobin-2008** [comment]
Hmm
"Merging is blocked
The base branch does not allow updates. [Learn more about protected branches.](https://docs.github.com/articles/about-protected-branches/)"

**Comment by davmac314** [comment]
Seems I locked the branch accidentally, I unlocked it now.

**Comment by mobin-2008** [comment]
Looks like it was a bug with github web, I merged with github mobile

**Pull request merged**